File: Archive/Stroika_FINAL_for_STERL_1992/Applications/NodeViewDemo/Sources/NodeViewDemo.cc
/*
 * $Header: /fuji/lewis/RCS/NodeViewDemo.cc,v 1.4 1992/09/08 16:48:46 lewis Exp $
 *
 * Description:
 *
 * TODO:
 *
 *
 * Notes:
 *
 * Changes:
 *	$Log: NodeViewDemo.cc,v $
 *		Revision 1.4  1992/09/08  16:48:46  lewis
 *		Renamed NULL -> Nil.
 *
 *		Revision 1.3  1992/09/01  17:31:57  sterling
 *		*** empty log message ***
 *
 *
 *
 *
 */


#include	"Debug.hh"
#include	"StreamUtils.hh"

#include	"Document.hh"
#include	"MenuBar.hh"

#include	"SimpleAboutBox.hh"

#include	"NodeViewCommands.hh"
#include	"NodeViewDemo.hh"
#include	"NodeViewDocument.hh"






NodeViewCommandNamesBuilder::NodeViewCommandNamesBuilder ()
{
	/*
	 * Add any command name to the command name table (database) that we define for
	 * our application.
	 */

#if		!qMPW_STATIC_ARRAY_WITH_GEN_INIT_CODE_GEN_BUG
	static	const
#endif
	CmdNameTablePair	kCommandNames[] = {
		CmdNameTablePair (CommandHandler::eAboutApplication, "About NodeViewDemo"),
		CmdNameTablePair (eConfigureNodes, "Configure Nodes"),
		CmdNameTablePair (eAddNode, "Add Node"),
		CmdNameTablePair (eRemoveNode, "Remove Nodes"),
		CmdNameTablePair (eShiftLeft, "Shift Left"),
		CmdNameTablePair (eShiftRight, "Shift Right"),
	};
	CommandNameTable::Get ().AddPairs (kCommandNames, sizeof (kCommandNames)/sizeof (kCommandNames[0]));
}


/*
 ********************************************************************************
 *************************************** NodeViewDemo ***************************
 ********************************************************************************
 */
NodeViewDemo*	NodeViewDemo::sThe	=	Nil;

NodeViewDemo::NodeViewDemo (int argc, const char* argv []) :
	Application (argc, argv),
	fMenuBar (),
	fSystemMenu (),
	fFileMenu (),
	fEditMenu ()
#if		qDebug
	,fDebugMenu ()
#endif
{
	Assert (sThe == Nil);
	sThe = this;

#if		qMacToolkit 
	fMenuBar.AddMenu (&fSystemMenu, fSystemMenu.DefaultName ());
#endif
	fMenuBar.AddMenu (&fFileMenu, fFileMenu.DefaultName ());
	
	fEditMenu.AddSeparatorMenuItem ();
	fEditMenu.AddStringMenuItem (eConfigureNodes);
	fEditMenu.AddStringMenuItem (eAddNode);
	fEditMenu.AddStringMenuItem (eRemoveNode);
	fEditMenu.AddStringMenuItem (eShiftLeft);
	fEditMenu.AddStringMenuItem (eShiftRight);
	
	fMenuBar.AddMenu (&fEditMenu, fEditMenu.DefaultName ());
#if		qDebug 
	fMenuBar.AddMenu (&fDebugMenu, fDebugMenu.DefaultName ());
#endif

	SetMenuBar (&fMenuBar);
}

NodeViewDemo::~NodeViewDemo ()
{
	SetMenuBar (Nil);
	
#if		qMacToolkit 
	fMenuBar.RemoveMenu (&fSystemMenu);
#endif
	fMenuBar.RemoveMenu (&fFileMenu);
	fMenuBar.RemoveMenu (&fEditMenu);
#if		qDebug 
	fMenuBar.RemoveMenu (&fDebugMenu);
#endif
}

void	NodeViewDemo::DoSetupMenus ()
{
	Application::DoSetupMenus ();

	EnableCommand (eAboutApplication, True);
	EnableCommand (eNew, True);
	EnableCommand (eOpen, True);
}

Boolean		NodeViewDemo::DoCommand (const CommandSelection& selection)
{
	switch (selection.GetCommandNumber ()) {
		case eAboutApplication:
			DisplaySimpleAboutBox ();
			return (True);

		case eNew:
			{
				NodeViewDocument*	theDoc	=	 new NodeViewDocument ();
				theDoc->DoNewState ();
			}
			return (True);

		case	eOpen: 
			{
				NodeViewDocument*	theDoc	=	 new NodeViewDocument ();
				Try {
					theDoc->Read ();
				}
				Catch () {
					// failed to open document, so delete it.
					theDoc->Close ();		// deletes it.
				}
			}
			return (True);
			
		default:
			return (Application::DoCommand (selection));
	}
	AssertNotReached (); return (True);
}





// For gnuemacs:
// Local Variables: ***
// mode:C ***
// tab-width:4 ***
// End: ***

File: Archive/Stroika_FINAL_for_STERL_1992/Applications/NodeViewDemo/Sources/NodeViewDemo.r
/* Copyright(c) Sophist Solutions Inc. 1990-1992.  All rights reserved */

/*
 * $Header: /fuji/lewis/RCS/NodeViewDemo.r,v 1.2 1992/09/01 17:32:16 sterling Exp $
 *
 * TODO:
 *
 * Changes:
 *	$Log: NodeViewDemo.r,v $
 *		Revision 1.2  1992/09/01  17:32:16  sterling
 *		*** empty log message ***
 *
 *
 *
 *
 */


#include	"ApplicationVersion"


resource 'vers' (1) {
	kMajorVersion,
	kMinorVersion,
	kStage,
	kStageVersion,
	verUS,
	kShortVersionString,
	kLongVersionString,
};


resource 'SIZE' (-1) {
	dontSaveScreen,
	acceptSuspendResumeEvents,
	enableOptionSwitch,
	canBackground,
	multiFinderAware,
	backgroundAndForeground,
	dontGetFrontClicks,
	ignoreChildDiedEvents,
	is32BitCompatible,
	notHighLevelEventAware,
	onlyLocalHLEvents,
	notStationeryAware,
	dontUseTextEditServices,
	reserved,
	reserved,
	reserved,
#if		qDebug
	1524	*	1024,						/* preferred mem size	*/
	1024		*	1024					/* minimum mem size		*/
#else
	750		*	1024,						/* preferred mem size	*/
	512		*	1024						/* minimum mem size		*/
#endif
};


type qCreator as 'STR ';
resource qCreator (0) {
	"NodeViewDemo"
};

resource 'BNDL' (128) {
	qCreator,
	0,
	{
		'ICN#',
		{
			0, 128
		},
		'FREF',
		{
			0, 128
		}
	}
};

resource 'FREF' (128) {
	'APPL',
	0,
	"Application"
};

resource 'ICN#' (128) {
	{	/* array: 2 elements */
		/* [1] */
		$"0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000"
		$"0000 0000 FE0E 0000 1010 0000 10E0 0C00 1080 1000 1080 20F8 1040 234C 107C 4040"
		$"1020 4040 1020 2040 0820 2040 0820 1040 003E 1040 0003 1040 0000 7040",
		/* [2] */
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
	}
};






/* For gnuemacs:
 * Local Variables: ***
 * mode:C ***
 * tab-width:4 ***
 * End: ***
 */

